screenpy_examples/screenpy/quietly_logging/actions/do_thing.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from screenpy import IsEqual, Performable, beat
+
+from screenpy_examples.screenpy.quietly_logging.actions.see import See
+from screenpy_examples.screenpy.quietly_logging.questions import SimpleQuestion
+
+if TYPE_CHECKING:
+    from screenpy import Actor
+
+
+class DoPassAfterAWhile(Performable):
+    LIMIT = 5
+    COUNTER = 0
+
+    @beat("{} tries to DoPassAfterAWhile")
+    def perform_as(self, actor: Actor):
+        if DoPassAfterAWhile.COUNTER > DoPassAfterAWhile.LIMIT:
+            DoPassAfterAWhile.COUNTER = 0
+            return
+
+        DoPassAfterAWhile.COUNTER += 1
+        raise AssertionError(f"This is going to Fail {DoPassAfterAWhile.COUNTER}")
+
+
+class DoFailCounter(Performable):
+    COUNTER = 0
+
+    @beat("{} tries to DoFailCounter")
+    def perform_as(self, actor: Actor):
+        DoFailCounter.COUNTER += 1
+        raise AssertionError(f"This is going to Fail {DoFailCounter.COUNTER}")
+
+
+class DoFail(Performable):
+    @beat("{} tries to DoFail")
+    def perform_as(self, actor: Actor):
+        actor.will(See(SimpleQuestion(), IsEqual(False)))
+
+
+class PerformPass(Performable):
```

Comment:
What's the logic between starting the Action with `Do` vs. `Perform`? We're in `do_thing.py` after all. :P